src/hotspot/share/runtime/vmThread.cpp

```diff
@@ -309,8 +218,10 @@ void VMThread::run() {
 // Notify the VMThread that the last non-daemon JavaThread has terminated,
 // and wait until operation is performed.
 void VMThread::wait_for_vm_thread_exit() {
-  assert(JavaThread::current()->is_terminated(), "Should be terminated");
-  { MonitorLocker mu(VMOperationQueue_lock, Mutex::_no_safepoint_check_flag);
+  assert(Thread::current()->is_Java_thread(), "Should be a JavaThread");
+  assert(((JavaThread*)Thread::current())->is_terminated(), "Should be terminated");
```

Comment:
Looks like just `assert(JavaThread::current()->is_terminated(), "Should be terminated")`.

Reply:
Fixing

Reply:
And delete the assert on L221.